### ISSUE TITLE
Pan gesture isn't working properly. 

### ISSUE THREAD

**Issue by AashishSapkota** [John-Lluch/SWRevealViewController#649]
Without adding pan gesture to 'front' ViewController, gesture gets activated after once toggling the rear view. (and works properly every time, after a tap on the 'rear' view toggle button.)
&
Adding pan gesture to 'front' ViewController, misplaces 'rear' ViewController margins. (and again works properly after a tap on the 'rear' view toggle button.)

Can someone please help..:)

**Comment by iDevelopper** [comment]
Try this. When you add the constraints to your rear view, uncheck Constrain to margins:

![2016-11-28_10-19-34](https://cloud.githubusercontent.com/assets/1782450/20662617/60e67bb4-b554-11e6-9ff4-a068bea955b3.png)

**Comment by AashishSapkota** [comment]
Yeah, I checked out constraints. But, I'm getting an error for only 'panGesture' until the sideBar button is tapped. Even 'panGesture' is working correctly, but only after a button is clicked.

Screenshot before the button is tapped for sideMenu;
<img width="221" alt="screen shot 2016-11-30 at 3 38 34 pm" src="https://cloud.githubusercontent.com/assets/6425249/20747826/aed3e206-b713-11e6-9225-a4646367a9bb.png">

Screenshot after the button is tapped for sideMenu;
<img width="222" alt="screen shot 2016-11-30 at 1 03 14 pm" src="https://cloud.githubusercontent.com/assets/6425249/20747834/b961279c-b713-11e6-86ff-7f50b2628abb.png">

**Comment by iDevelopper** [comment]
What kind of object is the top view in rear view. A navigation bar?

**Comment by iDevelopper** [comment]
I was able to fix the problem with an older version of Xcode ( see #579 ), but it seems that now is definitely a bug. You can have a look at my library and the example below (sorry John), with which you will not have this worry.

[SWLayout 2.zip](https://github.com/John-Lluch/SWRevealViewController/files/622017/SWLayout.2.zip)

**Comment by AashishSapkota** [comment]
Hey thanks  @iDevelopper, but I'm not getting you, coz i'm having problem with SWRevealView and I haven't used PBRevealView. 
Its just a UIView as top view in rear view. 

I couldn't even run your project, coz i'm with xcode-7.

**Comment by iDevelopper** [comment]
Ok, and Objective-c or Swift? I know you have an issue with SWRevealViewController!

**Comment by AashishSapkota** [comment]
Are you suggesting to change my SlideMenu library??

**Comment by iDevelopper** [comment]
No, but I can't resolve this issue. You could also embed your rear view in a navigation controller for replacing your view.

**Comment by iDevelopper** [comment]
I finally got it. Check yours constraints again. It works.

![2016-12-02_09-26-22](https://cloud.githubusercontent.com/assets/1782450/20827328/99afe8a2-b871-11e6-8fa2-7e1d3bdfeaa3.png)

**Comment by iDevelopper** [comment]
[SWLayoutXc7.zip](https://github.com/John-Lluch/SWRevealViewController/files/626717/SWLayoutXc7.zip)

**Comment by AashishSapkota** [comment]
I have two conditions here; 
 - If I add panGesture to 'front VC' than, above problem occurs,
 - If I don't add panGesture to 'front VC' than, panning can't be done to toggle SideMenu. (but only for first time, but if sideBar button is clicked once... panning can be done perfectly...as required.)

**Comment by iDevelopper** [comment]
Have you looked at my example? it works.

**Comment by AashishSapkota** [comment]
Yeah I've looked into your project @iDevelopper. But, even I'm not getting error while testing it in a new or a separate project. 
This project has already been built and its like 'ready for the release', so I'm searching for 'What could be the reason behind the issue?'.. so I could solve it with a better way. 
Isn't there any way to solve the above issue in a healthier way, maybe.
You have been trying to help me out with your valuable time, really appreciate that :-)

**Comment by iDevelopper** [comment]
This example was meant to show you that your constraints were not good. So you really have to look around for that.

**Comment by AashishSapkota** [comment]
If the constraints weren't correct than how would it be possible to work properly after a single tap gesture??
Why would it change its position?

**Plus** Have you seen those conditions I mentioned you in above comment??

**Comment by iDevelopper** [comment]
You can see that my sample works. But I had to go through it several times. It seems that this is a problem in Xcode. Try to put your constraints as you think and then modify them by editing them. Look at the constraints in my sample, and try to find somme differences.

**Comment by AashishSapkota** [comment]
I tried editing it with all your constraints, still facing the same problem. :(

**Comment by iDevelopper** [comment]
Is it possible for you to share a copy of your project with just the menu and the first front view controller (a light version)?
